crates/aptos-faucet/src/mint.rs

```diff
@@ -132,7 +132,20 @@ pub async fn process(service: &Service, params: MintParams) -> Result<Response>
     // We shouldn't have too many outstanding txns
     for _ in 0..60 {
         if our_faucet_seq < faucet_seq + 50 {
-            break;
+            // Enforce a stronger ordering of priorities based upon the MintParams that arrived
+            // first. Then put the other folks to sleep to try again until the queue fills up.
+            if service.outstanding_requests.read().unwrap().first() == Some(&params) {
+                // There might have been two requests with the same parameters, so we ensure that
+                // we only pop off one of them. We do a read lock first since that is cheap,
+                // followed by a write lock.
+                let mut requests = service.outstanding_requests.write().unwrap();
```

Comment:
where are you pushing/populating the `outstanding_requests`?

Reply:
muahaha

Reply:
HAHA, good reading... these things are inherently hard to test...